wallet/src/service/Offers.ts

```diff
@@ -41,6 +44,41 @@ type GiveOrWantEntries = {
   };
 };
 
+const CapDataShape = harden({ body: M.string(), slots: M.array() });
+
+const theOne = <T>(array: Array<T> | undefined) => {
+  assert(array?.length === 1);
+
+  const first = array.at(0);
+  assert(first !== undefined);
+
+  return first;
+};
+
+const getBrandPetNameAndDisplayInfoWithoutPurse = async (
+  watcher: ChainStorageWatcher,
+  brand: Brand,
+) => {
+  const brandsP = watcher.queryOnce<[string, unknown][]>([
+    AgoricChainStoragePathKind.Data,
+    'published.agoricNames.brand',
+  ]);
+
+  const [brands, boardAux] = await Promise.all([
+    brandsP,
+    theOne(
+      watcher.queryBoardAux<{
+        displayInfo: DisplayInfo;
+      }>([brand]),
+    ),
+  ]);
+
+  const brandPetname = theOne(brands.find(([_, brand]) => brand === brand));
```

Comment:
Did you test this before pushing? `brands` is an array of tuples of `[petname, brand]`, so `assert(array?.length === 1);` above would always throw since the length is 2. I think it's better to leave the `?.at(0)` and leave the `brandPetname` undefined if missing, then show an error in the component that renders the offer so that the user doesn't have to open the console to know something went wrong. Additionally, I think throwing will break the loop that watches pending offers, preventing it from rendering other offers.